doc/user-guide/Push-notifications.md

```diff
@@ -0,0 +1,204 @@
+}]}.
+
+{mod_push_service_mongoosepush, [
+    {pool_name, mongoose_push_http}
+    {api_version, "v1"}
+]}.
+```
+
+First, we create the HTTP pool for communicating with [MongoosePush][].
+Here, we assume that [MongoosePush][] will be available on localhost on port 8443 which is the default one.
+Next we enable [mod_push_service_mongoosepush][].
+First option is a name of the HTTP pool to use and the second one is a version of [MongoosePush][]'s API (currently only "_v1_" is supported).
+
+And that's it, thats the end of MongooseIM configuration.
+All we need to do now is to set up [MongoosePush][].
+
+### Starting [MongoosePush][]
+
+The easiest way to start [MongoosePush][] is using its [docker image](https://hub.docker.com/r/mongooseim/mongoose-push).
+But before you can set up [MongoosePush][], you need _FCM_ application token and/or _APNS_ application certificate.
+The _FCM_ token you can get [here](https://console.firebase.google.com/) and the easiest way of getting _APNS_ application certificate is by running [this](https://github.com/fastlane/fastlane/tree/master/pem) script (please note that you need certificate in `pem` format).
+
+After you get the _FCM_ application token and/or _APNS_ application certificate, you can prepare to start [MongoosePush][].
+Firstly, prepare the following files structure:
+
+* priv/
+    * ssl/
+        * rest_cert.pem - The REST endpoint certificate
+        * rest_key.pem - private key for the REST endpoint certificate
+    * apns/
+        * prod_cert.pem - Production APNS app certificate
+        * prod_key.pem - Production APNS app certificate's private key
+        * dev_cert.pem - Development APNS app certificate
+        * dev_jey.pem - Development APNS app certificate's private key
+
+Assuming that your _FCM_ app token is `MY_FCM_SECRET_TOKEN` and you have the `priv` directory with all ceriticates in current directory, then you may start MongoosePush with the following command:
+
+```bash
+docker run -v `pwd`/priv:/opt/app/priv \
+  -e PUSH_FCM_APP_KEY="MY_FCM_SECRET_TOKEN" \
+  -e PUSH_HTTPS_CERTFILE="ssl/rest_cert.pem" \
+  -e PUSH_HTTPS_KEYFILE="ssl/rest_key.pem" \
+  -it --rm mongooseim/mongoose-push:latest
+```
+
+If you don't want to use either _APNS_ or _FCM_, you simply need to pass `PUSH_APNS_ENABLED=0` or `PUSH_FCM_ENABLED=0` respectively as additional env variables in your `docker run` command.
+For more advanced options and configuration please refer to _"Quick start / Configuring"_ in [MongoosePush][]'s [_README.md_][MongoosePushReadme].
+
+When your [MongoosePush][] docker is up and running, Push Notifications now can be used in your MongooseIM instance.
+
+## Using push notifications on client side
+
+The XMPP client application has very few things to do in order to receive push notifications.
+The whole process described in this section along with example notification flow we can see in the following diagram:
+ ![](push_notifications_example_message_flow.svg)
```

Comment:
I think that's far better, I'm not sure it's natural though. Maybe it would be more natural if the step 2 was split into 2a and 2b, and this 2 lines doesn't have common point on MongooseIM.
@Nyco, @goddammit are you able to advice here?